modules/test/playwright/tests/client-extension-web/editorConfigContributor.spec.ts

```diff
@@ -128,20 +128,26 @@ test('Add a toolbar button to an Alloy Editor @LPD-11056', async ({
 			`${liferayConfig.environment.baseUrl}/web${site.friendlyUrlPath}${layout.friendlyUrlPath}`
 		);
 
-		await editorSamplePage.selectTab({tabLabel: 'Alloy'});
+		await expect(
+			editorSamplesPage.balloonEditorContainer.getByText('Lorem ipsum')
+		).toBeInViewport();
 
-		await editorSamplePage.alloyEditorContainer.isVisible();
+		await editorSamplesPage.selectTab({tabLabel: 'Alloy'});
+
+		await expect(
+			editorSamplesPage.alloyEditorContainer.getByText('Lorem ipsum')
+		).toBeInViewport();
 	});
 
 	await test.step('Check if client extenstion is applied', async () => {
-		await editorSamplePage.alloyEditorContainer
+		await editorSamplesPage.alloyEditorContainer
 			.getByText('Lorem ipsum')
 			.selectText();
 
-		const toolbarContainer = editorSamplePage.alloyEditorToolbarContainer;
-
-		await toolbarContainer.isVisible();
-
-		expect(toolbarContainer.getByTitle('Insert Video')).toBeVisible();
+		await expect(
+			editorSamplesPage.alloyEditorToolbarContainer.getByTitle(
+				'Insert Video'
+			)
+		).toBeInViewport();
```

Comment:
didn't `.toBeVisible()` work here? This opens a decision point when asserting visibility, it's a very interesting discussion

Reply:
I am not entirely sure on this. What I definitely did get is severall fake positives with `toBeVisible`. It checks size, display and opacity, but there is a lot CSS magic it seems to miss. What I am not sure is how effective is `toBeInViewport`. So far, for me, on a small sample size, it behaves better.